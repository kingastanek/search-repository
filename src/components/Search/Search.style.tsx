import styled from 'styled-components';

export const SearchWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const SearchInput = styled.input`
  border: 0;
  outline: 0;
  background: transparent;
  font-size: 1rem;
  border-bottom: 1px solid var(--theme-grey);
  padding: 0.5rem;
  margin-left: 0.5rem;
`;
