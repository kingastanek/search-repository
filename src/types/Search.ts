
export type tRepositoriesItemAPI = {
  name: string;
  owner: { login: string };
  created_at: Date;
  stargazers_count: number;
};

export type tRepositoriesData = {
  order: number;
  name: string;
  owner: string;
  stars: number;
  createdAt: Date;
};

export type tSearchParam = {
  q: string;
  per_page: number;
  page: number;
  sort?: string;
  order?: string;
};
