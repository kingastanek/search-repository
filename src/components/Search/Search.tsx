import React from 'react';
import { SearchWrapper, SearchInput } from './Search.style';

type tSearch = {
  name: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

const Search = ({ onChange, name, value }: tSearch): JSX.Element => {
  return (
    <SearchWrapper>
      <SearchInput
        name={name}
        value={value}
        onChange={onChange}
        placeholder='Search...'
      />
    </SearchWrapper>
  );
};

export default Search;
