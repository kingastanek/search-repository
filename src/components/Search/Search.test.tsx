import { shallow } from 'enzyme';
import Search from './Search';

describe('Search', () => {
  it('should render correctly - without value', () => {
    const component = shallow(
      <Search name='emptySearch' value='' onChange={() => jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render correctly - with value', () => {
    const component = shallow(
      <Search name='searchWithValue' value='test' onChange={() => jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
