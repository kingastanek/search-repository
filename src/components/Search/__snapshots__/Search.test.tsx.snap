// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly - with value 1`] = `ShallowWrapper {}`;

exports[`Search should render correctly - without value 1`] = `ShallowWrapper {}`;
